processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1010,6 +1024,17 @@ public QueryableIndex load(File inDir) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
+      Map<String, Object> metadata = null;
+      ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
+      if (metadataBB != null) {
+        metadata = mapper.readValue(
+            serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
```

Comment:
Can you catch/rethrow parse errors then, with more information that they came from the metadata? As it stands right now the message it will throw will probably not be helpful at all unless you dig into the source code.